### PR TITLE
Added life handling, initial bomb pwr, godmode and tests for these

### PR DESCRIPTION
**Pull request by Stocken99**
Initial Commit for a functional the player script. Added very basic handling of health, player death, initial bomb power and two tests which tests if the Method to kill works (should be improved later when this script is on a player), also tests if the GetALifeFunctions returns the correct health when godmode is activated.

#96 Lecture 9 Checklist
#98 Lecture 10 Checklist

### CODE REVIEW

File: Assets/GameLogic/CharacterMovement.cs [hunk 1]
```diff
@@ -2,8 +2,10 @@
 using Assets.GameLogic;
 using UnityEngine;
 
+
 public class CharacterMovement : MonoBehaviour
 {
+
```
Comment (MarrisSparrisNilsson) [inline Assets/GameLogic/CharacterMovement.cs:8]: Why add more code lines?

Comment (Stocken99) [reply Assets/GameLogic/CharacterMovement.cs:8]: Cause i love me some air in the morning <3

File: Assets/GameLogic/Player.cs [hunk 2]
```diff
@@ -1,21 +1,106 @@
 using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine;
 
 namespace GameLogic
 {
-    public abstract class Player
+
+    public class Player : MonoBehaviour
```
Comment (MarrisSparrisNilsson) [inline Assets/GameLogic/Player.cs:8]: Please rename this class to HumanPlayer.

File: Assets/GameLogic/Player.cs [hunk 3]
```diff
@@ -1,21 +1,106 @@
...
     {
         public List<Bomb> n_Bomb { get; set; }
+        public int Id { get; set; } // Do we need this?
+        public int bombStrengt { get; set; }
+        public int health { get; set; }
+
         public float speed { get; set; }
+
+        public bool isAlive { get; set; }
```
Comment (MarrisSparrisNilsson) [inline Assets/GameLogic/Player.cs:17]: These new properties that you've added will also have to go inside the interface that im working on.

File: Assets/GameLogic/Player.cs [hunk 4]
```diff
@@ -1,21 +1,106 @@
...
+            else if (isStart == 0) //increase pwr
+            {
+                return (bombStrengt += 1);
+            }
+            else return bombStrengt;
+        }
+
+
 
-        public abstract void PlaceBombRequest(Element[,]? gameBoard);
```
Comment (MarrisSparrisNilsson) [inline Assets/GameLogic/Player.cs:16]: A human player must also have to do these requests, they should not be deleted as of yet.

File: README.md [hunk 5]
```diff
@@ -1,4 +1,7 @@
+<<<<<<< HEAD
+=======
 [![Build and tests 🎮](https://github.com/Civil-Data/Bomb-IT/actions/workflows/actions.yml/badge.svg)](https://github.com/Civil-Data/Bomb-IT/actions/workflows/actions.yml)
+>>>>>>> c75dcf6bc72e0987640cb6f40112409902a08f95
```
Comment (MarrisSparrisNilsson) [inline README.md:4]: You have a merge conflict here?

**Review by MarrisSparrisNilsson — CHANGES_REQUESTED** [review]
Make the following changes that I've stated. But overall good job!

File: Assets/GameLogic/Player.cs (hunk 2 above)
Comment (Stocken99) [reply Assets/GameLogic/Player.cs]: Fixed

**Review by MarrisSparrisNilsson — APPROVED** [review]

**Pull request merged**